From 007095f6020b30b32e9cde6cf5826b7300749e19 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Fri, 7 Mar 2014 13:26:16 +0000
Subject: [PATCH 056/118] java: run: Add java/.libs to LD_LIBRARY_PATH so JVM
 finds the right JNI file.

(cherry picked from commit 6aa64ee3a3f1c990b5f9ce537f2b840273c3c1ba)
---
 java/run-bindtests  | 2 +-
 java/run-java-tests | 2 +-
 run.in              | 4 ++--
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/java/run-bindtests b/java/run-bindtests
index 8a60df2..ad418d1 100755
--- a/java/run-bindtests
+++ b/java/run-bindtests
@@ -18,5 +18,5 @@
 
 set -e
 
-$JAVA_EXE -Djava.library.path=.libs Bindtests > bindtests.tmp
+$JAVA_EXE Bindtests > bindtests.tmp
 diff -u $srcdir/../bindtests bindtests.tmp
diff --git a/java/run-java-tests b/java/run-java-tests
index 1d3c66a..268b6c9 100755
--- a/java/run-java-tests
+++ b/java/run-java-tests
@@ -20,5 +20,5 @@ set -e
 
 for f in $(ls -1 t/*.class | fgrep -v \$); do
   classname=$(basename $f .class)
-  $JAVA_EXE -Djava.library.path=.libs -ea $classname
+  $JAVA_EXE -ea $classname
 done
diff --git a/run.in b/run.in
index a642d14..873b230 100755
--- a/run.in
+++ b/run.in
@@ -66,9 +66,9 @@ chcon --reference=/tmp tmp 2>/dev/null ||:
 export LIBGUESTFS_PATH="$b/appliance"
 
 if [ -z "$LD_LIBRARY_PATH" ]; then
-    LD_LIBRARY_PATH="$b/src/.libs:$b/gobject/.libs"
+    LD_LIBRARY_PATH="$b/src/.libs:$b/java/.libs:$b/gobject/.libs"
 else
-    LD_LIBRARY_PATH="$b/src/.libs:$b/gobject/.libs:$LD_LIBRARY_PATH"
+    LD_LIBRARY_PATH="$b/src/.libs:$b/java/.libs:$b/gobject/.libs:$LD_LIBRARY_PATH"
 fi
 export LD_LIBRARY_PATH
 
-- 
2.7.4

